component/src/main/java/org/wso2/carbon/identity/authenticator/reddit/RedditAuthenticator.java

```diff
@@ -0,0 +1,296 @@
+import org.wso2.carbon.identity.application.authentication.framework.exception.AuthenticationFailedException;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.application.authenticator.oidc.OIDCAuthenticatorConstants;
+import org.wso2.carbon.identity.application.authenticator.oidc.OpenIDConnectAuthenticator;
+import org.wso2.carbon.identity.application.common.model.ClaimMapping;
+import org.wso2.carbon.identity.application.common.model.Property;
+import org.wso2.carbon.identity.application.common.util.IdentityApplicationConstants;
+import org.wso2.carbon.identity.base.IdentityConstants;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Authenticator of Reddit
+ */
+public class RedditAuthenticator extends OpenIDConnectAuthenticator implements FederatedApplicationAuthenticator {
+
+    private static Log log = LogFactory.getLog(RedditAuthenticator.class);
+
+    /**
+     * Get Reddit authorization endpoint.
+     */
+    @Override
+    protected String getAuthorizationServerEndpoint(Map<String, String> authenticatorProperties) {
+        return RedditAuthenticatorConstants.REDDIT_OAUTH_ENDPOINT;
+    }
+
+    /**
+     * Get Reddit token endpoint.
+     */
+    @Override
+    protected String getTokenEndpoint(Map<String, String> authenticatorProperties) {
+        return RedditAuthenticatorConstants.REDDIT_TOKEN_ENDPOINT;
+    }
+
+    /**
+     * Get Reddit user info endpoint.
+     */
+    @Override
+    protected String getUserInfoEndpoint(OAuthClientResponse token, Map<String, String> authenticatorProperties) {
+        return RedditAuthenticatorConstants.REDDIT_USERINFO_ENDPOINT;
+    }
+
+    /**
+     * Check ID token in Reddit OAuth.
+     */
+    @Override
+    protected boolean requiredIDToken(Map<String, String> authenticatorProperties) {
+        return false;
+    }
+
+    /**
+     * Get the friendly name of the Authenticator
+     */
+    @Override
+    public String getFriendlyName() {
+        return RedditAuthenticatorConstants.AUTHENTICATOR_FRIENDLY_NAME;
+    }
+
+    /**
+     * Get the name of the Authenticator
+     */
+    @Override
+    public String getName() {
+        return RedditAuthenticatorConstants.AUTHENTICATOR_NAME;
+    }
+
+    /**
+     * Get the scope
+     */
+    public String getScope(String scope, Map<String, String> authenticatorProperties) {
+        scope = authenticatorProperties.get(RedditAuthenticatorConstants.SCOPE);
+        if (StringUtils.isEmpty(scope)) {
+            scope = RedditAuthenticatorConstants.USER_SCOPE;
+        }
+        return scope;
+    }
+
+
```

Comment:
Fixed in https://github.com/wso2-extensions/identity-outbound-auth-reddit/pull/3